sample/normal.go

```diff
@@ -19,60 +19,136 @@ package sample
 import (
 	"math"
 	"math/big"
-	"math/rand"
 )
 
 // Normal samples random values from the Normal (Gaussian)
 // probability distribution, centered on 0.
 type Normal struct {
```

Comment:
As we discussed, `Normal` struct represents the common skeleton used by other discrete gaussian distribution. By itself it's not a sampler anymore - (it doesn't implement `Sampler` interface), so we need to make it unexported, e.g. `normal`, to prevent users/other packages from using it. Similarly, `NewNormal` should be `newNormal`.

PS: If we can come up with a better, more descriptive name for this struct (one that actually reflects its purpose), I'll be more than happy :smile:

Reply:
No good idea for a new name...